waitress/channel.py

```diff
@@ -305,6 +308,12 @@ def del_channel(self, map=None):
     #
 
     def write_soon(self, data):
+        if not self.connected:
+            # if the socket is closed then interrupt the task so that it
+            # can cleanup possibly before the app_iter is exhausted
+            # XXX a simple boolean check should not require a lock since it's
```

Comment:
s/XXX/NB/ and doing an if on a boolean here is safe in CPython, and probably other interpreters, and there is no race.

I don't think this comment is necessary to be honest.